src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/dormant/DormantEventsAboutToSubmitHandler.java

```diff
@@ -46,6 +66,46 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
         caseData.setDirectionDueDate(null);
         callback.getCaseDetailsBefore().ifPresent(f -> caseData.setPreviousState(f.getState()));
 
+        String caseID = callback.getCaseDetails().getCaseData().getCcdCaseId();
+        if (eligibleForHearingsCancel.test(callback)) {
+
+            CancellationReason cancellationReason = null;
+            switch (callback.getEvent()) {
+                case CONFIRM_LAPSED:
+                case HMCTS_LAPSE_CASE:
+                case ADMIN_SEND_TO_DORMANT_APPEAL_STATE:
+                case LAPSED_REVISED:
+                    cancellationReason = CancellationReason.LAPSED;
+                    break;
+                case WITHDRAWN:
+                    cancellationReason = CancellationReason.WITHDRAWN;
+                    break;
+                default: log.info("CaseID: {} - Event: {} is not handled for cancellation reason",
+                        caseID, callback.getEvent());
+                    break;
+            }
+
+            log.info("CaseID: {} - Event: {}. HearingRoute is List Assist - Sending cancellation message",
+                    caseID, callback.getEvent());
+            hearingMessageHelper.sendListAssistCancelHearingMessage(caseID, cancellationReason);
+        }
+
         return new PreSubmitCallbackResponse<>(caseData);
     }
+
+    private final Predicate<Callback<SscsCaseData>> eligibleForHearingsCancel = callback ->
+            SscsUtil.isValidCaseState(callback.getCaseDetailsBefore().map(CaseDetails::getState)
+                    .orElse(State.UNKNOWN), List.of(State.HEARING, State.READY_TO_LIST))
+            && SscsUtil.isSAndLCase(callback.getCaseDetails().getCaseData())
+            && isValidCaseEventForCancellation(callback.getEvent())
+            && isScheduleListingEnabled;
+
+    private boolean isValidCaseEventForCancellation(EventType event) {
+        return List.of(
+                CONFIRM_LAPSED,
+                HMCTS_LAPSE_CASE,
+                ADMIN_SEND_TO_DORMANT_APPEAL_STATE,
+                LAPSED_REVISED,
+                WITHDRAWN).contains(event);
```

Comment:
Minor: Would be better to have the list as a constant than creating it every time. I'd guess the jvm would optimise anyway, but still good practice all the same